src/shared/codelens/codeLensUtils.ts

```diff
@@ -147,37 +135,26 @@ export async function makeCSharpCodeLensProvider(): Promise<vscode.CodeLensProvi
                 document,
                 handlers,
                 token,
-                language: 'csharp',
             })
         },
     }
-
-    return codeLensProvider
 }
 
 export function makeTypescriptCodeLensProvider(): vscode.CodeLensProvider {
+    const logger = getLogger()
+
     return {
         provideCodeLenses: async (
             document: vscode.TextDocument,
             token: vscode.CancellationToken
         ): Promise<vscode.CodeLens[]> => {
-            const search: TypescriptLambdaHandlerSearch = new TypescriptLambdaHandlerSearch(
-                document.uri.fsPath,
-                document.getText()
-            )
-            const unprocessedHandlers: LambdaHandlerCandidate[] = await search.findCandidateLambdaHandlers()
-
-            // For Javascript CodeLenses, store the complete relative pathed handler name
-            // (eg: src/app.handler) instead of only the pure handler name (eg: app.handler)
-            // Without this, the CodeLens command is unable to resolve a match back to a sam template.
-            // This is done to address https://github.com/aws/aws-toolkit-vscode/issues/757
-            const handlers = await tsDebug.decorateHandlerNames(unprocessedHandlers, document.uri)
+            const handlers = await tsDebug.getLambdaHandlerCandidates(document)
+            logger.debug('makeTypescriptCodeLensProvider handlers:', JSON.stringify(handlers, undefined, 2))
```

Comment:
I believe this will log on most all character edits to a file. That might be a bit spammy even for debug level. Are you sure you want to keep this log in?

Reply:
We've got it for the other handlers. Basically, I wanted to make the `makeXCodeLensProviders` as similar as possible with this commit. I'm open to swapping that.

Alternately, we could try using something like lodash debounce: https://lodash.com/docs/#debounce

Reply:
Thanks for the explanation. Lets go for similarity now, and assess it once all the changes are complete.